src/class-wc-gateway-btcpay.php

```diff
@@ -621,7 +627,7 @@ public function process_payment($order_id)
                 throw new \Exception('The BTCPay payment plugin was called to process a payment but could not retrieve the order details for order_id ' . $order_id . '. Cannot continue!');
             }
 
-            $notification_url = $this->get_option('notification_url', WC()->api_request_url('WC_Gateway_BtcPay'));
+            $notification_url = $this->get_option('notification_url', get_home_url() . '/wp-json/btcpay/ipn/status');
```

Comment:
What will happen if the user have set `notification_url`, then we update?
I expect his store will break, because the link set was the old `WC_Gateway_BtcPay`.

What about using `/wc-api/WC_Gateway_BtcPay/` instead of `/wp-json/btcpay/ipn/status` as suffix for the route so things do not break?

Reply:
/wp-json is the default path when adding custom end points, see https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/

Better safe than sorry: I'll revert to the old style for the IPNs. I'll update the PR.

Reply:
Another possibility is having both.
Keeping the code as you did in this PR, but registering `/wc-api/WC_Gateway_BtcPay` as well?

Reply:
I'd rather only include stuff that's needed now (include country, remove strict order check, small syntax fix). The IPN processor change can be done in a different PR.

Please check the updated PR.